storage/repository.go

```
@@ -0,0 +1,152 @@
+// fail with an error. If a Copy implementation is already registered for this
+// scheme, it will be replaced with this one silently.
+//
+// If no Copy implementation is registered for a particular scheme, then copies
+// will be implemented as if the URIs were of different types, that is the
+// source URI will be read using ReaderFrom(), and the destination written
+// using WriterTo(). This function is an opportunity to implement a more
+// optimal approach, such as leveraging remote operations for a network-backed
+// repository.
+//
+// Since 2.0.0
+func RegisterCopy(scheme string, copyImplementation func(fyne.URI, fyne.URI) error) error {
+	// TODO
+}
+
+// RegisterRename registers an implementation of a Rename() function for a
+// specific URI scheme. This function should only be called in
+// Repository.Init(). If the scheme does not have a registered repository, then
+// this function will fail with an error. If a Rename implementation is already
+// registered for this scheme, it will be replaced with this one silently.
+//
+// If no Rename implementation is registered for a particular scheme, then
+// renames will be implemented as if the URIs were of different types, that is
+// the source URI will be read using ReaderFrom(), and the destination written
+// using WriterTo(), then the source will be deleted using Delete(). This
+// function is an opportunity to implement a more optimal approach, such as
+// leveraging remote operations for a network-backed repository.
+//
+// Since 2.0.0
+func RegisterRename(scheme string, renameImplementation func(fyne.URI, fyne.URI) error) error {
+}
+
+// RegisterRepository registers a storage repository so that operations on URIs
+// of the registered scheme will use methods implemented by the relevant
+// repository implementation. This method will call repository.Init(), and may
+// error if it does.
+//
+// Since 2.0.0
+func RegisterRepository(scheme string, repository Repository) error {
```

Will this will also silently replace any previously registered Repository? Will the previously registered Repository get any notification that it is no longer in use?

---

From https://github.com/fyne-io/fyne/wiki/Proposal:-The-Fyne-URI-Philosophy#revised-proposal; yes and yes via `repository.Destoy()`